omnigan/losses.py

```diff
@@ -276,6 +276,23 @@ def __call__(self, prediction, target):
         relDiff = torch.mean(d) * torch.mean(d)
         return diff - relDiff
 
+class SIGMLoss(nn.Module):
+    def __init__(self, gmweight):
+        super(SIGMLoss, self).__init__()
+        self.gmweight = gmweight
+        sobelx = torch.Tensor([[1,0,-1],[2,0,-2],[1,0,-1]])
+        sobely = torch.Tensor([[1,2,1],[0,0,0],[-1,-2,-1]])
+        self.filter = (0.5 * (sobelx + sobely))
+        
+    def __call__(self, prediction, target):
+        # get gradient map with sobel filters
+        batch_size = prediction.size()[0]
+        self.filter = (self.filter).expand((batch_size, 1, -1, -1)).to(device='cuda')      
+        grad_pred = F.conv2d(prediction, self.filter, stride=1)
+        grad_target = F.conv2d(target, self.filter, stride=1)
+        GMLoss = self.gmweight * torch.norm(grad_pred - grad_target)
+        diff = SIMSELoss()(prediction, target) + GMLoss
```

Comment:
if `SIMSELoss` is initialized at every call, you might as well move this to the init (`self.simse_loss = SIMSELoss()`) and then in the call: `diff = self.simse_loss(prediction, target) + gm_loss`